prow/plank/controller.go

```diff
@@ -529,7 +529,13 @@ func (c *Controller) startPod(pj *prowapi.ProwJob) error {
 	if err != nil {
 		return err
 	}
-	pod.Namespace = c.config().PodNamespace
+
+	// If the prowJob has a namespace specified use that value, else use the config prowJob namespace.
+	if pj.Spec.Namespace != "" {
```

Comment:
IMHO we should do this during config load and not in individual components

Reply:
Works for me... Let me try to move this to config.